lib/home_page.dart

```diff
@@ -8,15 +24,275 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
+  //needs to be initialized in case the subscription gets something new before load is done
+  List<Model> _items = <Model>[];
+  bool _fullyLoaded = false;
+
+  @override
+  void initState() {
+    load();
+    final int profileId = SupabaseManager.getCurrentProfile()!.id!;
+    SupabaseManager.supabaseClient.channel('public:messages').on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'INSERT', schema: 'public', table: 'messages', filter: 'sender_id=neq.$profileId'),
+      (payload, [ref]) {
+        if (payload['new']['sender_id'] != profileId) {
+          _loadNewMessage(payload['new']['id']);
+        }
+      },
+    ).on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'UPDATE', schema: 'public', table: 'messages', filter: 'sender_id=neq.$profileId'),
+      (payload, [ref]) {
+        if (payload['new']['read'] == true) {
+          setState(() {
+            _items =
+                _items.where((Model element) => element is RideEvent || element.id != payload['new']['id']).toList();
+          });
+        }
+      },
+    ).subscribe();
+    SupabaseManager.supabaseClient.channel('public:ride_events').on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'INSERT', schema: 'public', table: 'ride_events'), //, filter: 'sender_id=neq.$profileId'
+      (payload, [ref]) {
+        if (payload['new']['sender_id'] != profileId) {
+          _loadNewRideEvent(payload['new']['id']);
+        }
+      },
+    ).on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'UPDATE', schema: 'public', table: 'ride_events'), //, filter: 'sender_id=neq.$profileId'
+      (payload, [ref]) {
+        if (payload['new']['read'] == true) {
+          setState(() {
+            _items = _items.where((Model element) => element is Message || element.id != payload['new']['id']).toList();
```

Comment:
I think you might be looking for  `removeWhere`: https://api.flutter.dev/flutter/dart-core/List/removeWhere.html

Reply:
thanks for the hint